bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/AbstractMQTTThingHandler.java

```diff
@@ -105,8 +107,14 @@ public AbstractMQTTThingHandler(Thing thing, int subscribeTimeout) {
      * @return A future that completes normal on success and exceptionally on any errors.
      */
     protected CompletableFuture<@Nullable Void> start(MqttBrokerConnection connection) {
-        return availabilityStates.values().parallelStream().map(cChannel -> cChannel.start(connection, scheduler, 0))
-                .collect(FutureCollector.allOf());
+        @NonNull
+        Collector<@NonNull CompletableFuture<@Nullable Void>, @NonNull Set<@NonNull CompletableFuture<@Nullable Void>>, @NonNull CompletableFuture<@Nullable Void>> allOfCollector = FutureCollector
+                .allOf();
+        return availabilityStates.values().stream().map(cChannel -> {
+            final @NonNull CompletableFuture<@Nullable Void> fut;
+            fut = cChannel == null ? CompletableFuture.completedFuture(null) : cChannel.start(connection, scheduler, 0);
+            return fut;
+        }).collect(allOfCollector);
```

Comment:
`cChannel.start()` has some bits `synchronized` anyways, so this is good change.

Other changes are to remove null typing warnings